torchvision/ops/boxes.py

```diff
@@ -325,13 +325,13 @@ def complete_box_iou(boxes1: Tensor, boxes2: Tensor, eps: float = 1e-7) -> Tenso
 
     diou, iou = _box_diou_iou(boxes1, boxes2, eps)
 
-    w_pred = boxes1[:, 2] - boxes1[:, 0]
-    h_pred = boxes1[:, 3] - boxes1[:, 1]
+    w_pred = boxes1[:, None, 2] - boxes1[:, None, 0]
```

Comment:
why this is needed?

Reply:
This is necessary for the broadcasting to work properly. See also:
https://github.com/pytorch/vision/blob/ea0be26b88778b1033d4a176be68bcdd008ff934/torchvision/ops/boxes.py#L368-L371